src/iop/crop.c

```diff
@@ -1111,6 +1117,37 @@ static gchar *_aspect_format(gchar *original,
     return g_strdup_printf("%s  %4.2f", original, (float)adim / (float)bdim);
 }
 
+static void _crop_handle_flip(dt_iop_module_t *self, const dt_image_orientation_t mode)
+{
+  dt_iop_crop_params_t *p = self ? self->params : NULL;
+  if(!p) return;
+
+  // we avoid action if default or fully symmetrical
+  const gboolean no_action = (p->cx == 1.f - p->ch) && (p->cy == 1.f - p->cw);
```

Comment:
?

This is not testing for "full symmetry". ;-)

Also, some layouts that don't need adjustment under flipping do need to be rotated and vice versa. So if you want to do more than only avoid adjusting no-crops, you'll have to do a full before-after comparison, i.e. 
```
const float ocx = p->cx, ocy = p->cy, ocw = p->cw, och = p->ch;
if(mode == ORIENTATION_FLIP_HORIZONTALLY)        {p->cx = 1.f-ocw; p->cw = 1.f-ocx;}
  else if(mode == ORIENTATION_FLIP_VERTICALLY)   {p->cy = 1.f-och; p->ch = 1.f-ocy;}
  else if(mode == ORIENTATION_ROTATE_CW_90_DEG)  {p->cx = 1.f-och; p->ch =     ocw; p->cw = 1.f-ocy; p->cy =     ocx;}
  else if(mode == ORIENTATION_ROTATE_CCW_90_DEG) {p->cx =     ocy; p->cy = 1.f-ocw; p->cw =     och; p->ch = 1.f-ocx;}
```
and skip the last two lines if all of p->cx vs ocx etc are sigma close enough (since these are calculated values, rather than constants like 0 or 1, you could get mismatches on 1/4 == 1 - 3/4 if someone types in 25% everywhere).

Reply:
I'll stay with "avoid no-crops"